src/fontra/views/editor/editor.js

```diff
@@ -1780,6 +1845,175 @@ export class EditorController {
     });
   }
 
+  canAddAnchor() {
+    return this.sceneModel.getSelectedPositionedGlyph()?.glyph.canEdit;
+  }
+
+  async doAddAnchor() {
+    const point = this.sceneController.selectedGlyphPoint(this.contextMenuPosition);
+    const { anchor: tempAnchor } = await this.doAddEditAnchorDialog(undefined, point);
+    if (!tempAnchor) {
+      return;
+    }
+
+    let newAnchor = {
+      name: tempAnchor.name ? tempAnchor.name : "anchorName",
+      x: tempAnchor.x ? tempAnchor.x : Math.round(point.x),
+      y: tempAnchor.y ? tempAnchor.y : Math.round(point.y),
+    };
+    const instance = this.sceneModel.getSelectedPositionedGlyph().glyph.instance;
+    let relativeScaleX;
+    if (instance.xAdvance != 0) {
+      relativeScaleX = Math.round(point.x) / instance.xAdvance;
+    }
+
+    await this.sceneController.editLayersAndRecordChanges((layerGlyphs) => {
+      for (const layerGlyph of Object.values(layerGlyphs)) {
+        if (!tempAnchor.x && layerGlyph.xAdvance != 0) {
```

Comment:
```js
        if (relativeScaleX != null) {
          newAnchor.x = layerGlyph.xAdvance * relativeScaleX;
        }
```

(Not sure why you check `!tempAnchor.x`)

Reply:
Thanks for pointing me to this. The idea was correct, but the solution was wrong. tempAnchor.x could be also 0, in that case it would be wrong. We still need to check if tempAnchor.x is a valid number, because if that is the case, then this values would be used and not the calculated one.
Thanks! I fixed it.

Reply:
I still don't understand the need for checking tempAnchor.x. I currently get an unrounded x value if I place a new anchor.

There is something very unclear about this code, and I cannot figure out its intention.

Reply:
I fixed the unrounded x.